### PR TITLE
add getRealmRoles and getResourceRoles methods

### PR DESCRIPTION
**Pull request by rachael-oregan**
## Motivation

Github Issue #103 

## Description

Added `getRealmRoles()` and `getResourceRoles()` methods to improve the user experience of getting different users roles.

### CODE REVIEW

File: auth/src/main/java/org/aerogear/mobile/auth/user/UserPrincipalImpl.java [hunk 1]
```diff
@@ -128,6 +145,20 @@ public Builder withRoles(final Set<UserRole> roles) {
...
         }
 
+        public Builder withRealmRoles(final Set<UserRole> realmRoles) {
+            if (realmRoles != null) {
+                this.realmRoles.addAll(realmRoles);
+            }
+            return this;
+        }
+
+        public Builder withResourceRoles(final Set<UserRole> resourceRoles) {
```
Comment (wei-lee) [inline auth/src/main/java/org/aerogear/mobile/auth/user/UserPrincipalImpl.java]: I don't think we need those methods. They can be retrieved from the existing `roles` set. They will have different `RoleType` in the roles set.

**Comment by rachael-oregan** [comment]
@wei-lee I refactored to remove those builder methods and fields for the specific roles.  Let me know if its good to merge. Thanks :)

File: auth/src/main/java/org/aerogear/mobile/auth/user/UserPrincipalImpl.java [hunk 2]
```diff
@@ -206,6 +206,42 @@ public String getName() {
...
+        return realmRoles;
+    }
+
+    /**
+     * Get's the user resource roles
+     *
+     * @return user's resource roles
+     */
+    @Override
+    public Set<UserRole> getResourceRoles() {
```
Comment (wei-lee) [inline auth/src/main/java/org/aerogear/mobile/auth/user/UserPrincipalImpl.java]: those 2 methods are the same, except the RoleType. We can make that a parameter and a single method can used by both methods here.

Comment (rachael-oregan) [reply auth/src/main/java/org/aerogear/mobile/auth/user/UserPrincipalImpl.java]: Added this single method there

**Pull request merged**